### DISCUSSION TITLE
How to fix such interesting effect 😁

### DISCUSSION THREAD

**Discussion by MaximSaveliev** [tailwindlabs/tailwindcss#10728] [Help] [answered]
I have a Home with different sections also have some element witch appear above the main content such as search bar.
Sections with relative and absolute classes has wrong behaver. They are not hidden when search bar appears.
[Code on TailwindPlay](https://play.tailwindcss.com/pxan8GeBvE)
**Examples how it works**
![image](https://user-images.githubusercontent.com/60519187/222981585-c8960b6c-7f54-4d9b-9458-bce80b371424.png)
![image](https://user-images.githubusercontent.com/60519187/222981618-298998ec-bbb3-4e1d-8455-c8bad4046172.png)

**Comment by wongjn** [comment, marked as answer]
`absolute` and `relative` classes apply `position: absolute` and `position: relative` respectively. Neither of these would necessarily hide an element it is applied to. To hide an element with CSS, you could:
- Apply `display: none` with the `hidden` class.
- Apply `opacity: 0` with the `opacity-0` class.
- Apply `visibility: hidden` with the `invisible` class.
- Apply `sr-only` class (to hide the element while still making it available to assistive technology).

**Reply by MaximSaveliev** [reply]
Thanks, I know how to hide elements but i wanna save them on a page without ```hidden``` class.
 
 Orange part is correct for me, but violet rectangle is incorrect.
 I wanna make everything as a orange rectangle.

![Image](https://user-images.githubusercontent.com/60519187/222983610-0d072351-1c47-47aa-a5fd-87038e70946d.png)

**Reply by wongjn** [reply]
Ah OK, so you're having trouble with stacking order of elements. Consider reviewing the concept of [stacking contexts](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context).

**Reply by MaximSaveliev** [reply]
As I understand, I should play with ```z-...``` class.
Should I change it for whole page or i can make with that two card, because I changed ```z-10,20,30,50``` class but it doesn't work.
Although on right-side menu ```z-index``` = 100 🤔

**Reply by wongjn** [reply]
It seems like you have some different stacking contexts within the page such that the `z-index`es are only relative to the other elements within the same stacking context. You'd need to review the elements that are in the same stacking context and resolve their stacking order when in the same stacking context. Consider reviewing the concept of [stacking contexts](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context).

**Reply by MaximSaveliev** [reply]
Thanks a lot 🥳🥳🥳
Now it works 🎉🎉🎉🎉🎉
![image](https://user-images.githubusercontent.com/60519187/222988340-1d44b8c2-699f-4826-82fd-718d89df48be.png)
